test/e2e/es_cluster_logging.go

```diff
@@ -0,0 +1,324 @@
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"encoding/json"
+	"fmt"
+	"math"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/api"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/client"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/labels"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Cluster level logging using Elasticsearch", func() {
+	var c *client.Client
+
+	BeforeEach(func() {
+		var err error
+		c, err = loadClient()
+		Expect(err).NotTo(HaveOccurred())
+	})
+
+	It("should check that logs from pods on all nodes are ingested into Elasticsearch", func() {
+		ClusterLevelLoggingWithElasticsearch(c)
+	})
+})
+
+func bodyToJSON(body []byte) (map[string]interface{}, error) {
+	var r map[string]interface{}
+	if err := json.Unmarshal(body, &r); err != nil {
+		Logf("Bad JSON: %s", string(body))
+		return nil, fmt.Errorf("failed to unmarshal Elasticsearch response: %v", err)
+	}
+	return r, nil
+}
+
+// ClusterLevelLoggingWithElasticsearch is an end to end test for cluster level logging.
+func ClusterLevelLoggingWithElasticsearch(c *client.Client) {
+
+	// TODO: For now assume we are only testing cluster logging with Elasticsearch
+	// on GCE. Once we are sure that Elasticsearch cluster level logging
+	// works for other providers we should widen this scope of this test.
+	if testContext.provider != "gce" {
+		Logf("Skipping cluster level logging test for provider %s", testContext.provider)
+		return
+	}
+
+	// Check for the existance of the Elasticsearch service.
+	By("Checking the Elasticsearch service exists.")
+	const ns = api.NamespaceDefault
+	s := c.Services(ns)
+	// Make a few attempts to connect. This makes the test robust against
+	// being run as the first e2e test just after the e2e cluster has been created.
+	var err error
+	for attempt := 1; attempt < 10; attempt++ {
+		if _, err = s.Get("elasticsearch-logging"); err == nil {
+			break
+		}
+		Logf("Attempt %d to check for the existence of the Elasticsearch service failed", attempt)
+		time.Sleep(2 * time.Second)
+	}
+	Expect(err).NotTo(HaveOccurred())
+
+	// Wait for the Elasticsearch pods to enter the running state.
+	By("Checking to make sure the Elasticsearch pods are running")
+	label := labels.SelectorFromSet(labels.Set(map[string]string{"name": "elasticsearch-logging"}))
+	pods, err := c.Pods(ns).List(label)
+	Expect(err).NotTo(HaveOccurred())
+	for _, pod := range pods.Items {
+		err = waitForPodRunning(c, pod.Name)
+		Expect(err).NotTo(HaveOccurred())
+	}
+
+	// Query against the root URL for Elasticsearch.
+	body, err := c.Get().Prefix("proxy").Resource("services").Name("elasticsearch-logging").DoRaw()
```

Comment:
The c.Get() below (line 142) is broken across lines and quite a bit easier to read. Please make them consistent.